orderBook.js

```diff
@@ -0,0 +1,34 @@
+function reconcileOrder(existingBook, incomingOrder) {
+  const oppOrderTypes = existingBook.filter(order => order.type !== incomingOrder.type)
+  let updatedBook = existingBook.filter(order => order.type === incomingOrder.type)
+  updatedBook = oppOrderTypes.length
```

Comment:
If you're gonna short circuit on the existingBook.length you should do that before you start filtering.